HACKING.md

```diff
@@ -1,162 +1,193 @@
-
+# Hacking on the Cluster Logging Operator
+
+## Preparation
+
+Clone this repository into `$GOPATH/github.com/openshift/cluster-logging-operator`
+
+Install `podman`. For example on  Fedora:
+```
+sudo dnf install -y podman
+```
+
+Check out some other required repositories in your $GOPATH:
+```
+go get -d -u github.com/openshift/elasticsearch-operator/... github.com/openshift/origin-aggregated-logging/...
+```
+
+**Note:** `podman` copies our source tree very slowly. You can use `docker`
+instead, this will not exactly reproduce how images are built for release or by
+CI, but is close enough for many purposes. Put a symlink early in your path:
+```
+ln -s $(which docker) /somewhere/early/in/my/PATH/podman`
+```
+
+## Testing Overview
+
+There are two types of test:
+
+* *Unit tests:* run in seconds, no cluster needed, verify internal behavior - behind the API. New code *should* be reasonably well covered by unit tests but we don't have a formal requirement.
+* *E2E tests:* run in minutes to hour(s), requires a cluster. There are several ways to run these:
+   - *CI tests:* run automatically when you raise a PR in a controlled cluster environment. Can take hour(s) to get results.
+   - *Deployed tests:* Build and deploy CLO image to your own cluster, allows you to run tests selectively.
+   - *Local process:* requires a cluster with dependencies deployed (elasticsearch), but runs CLO as a *local process*. Faster and easier to debug.
+
+All PRs should pass unit tests. CI runs all tests automatically and won't merge a PR till they pass, so it is OK to create a PR and let CI run the e2e tests if you've done enough testing to be confident. This allows for human review and feedback while CI tests are running.
+
+To get a code review of unfinished work, create a PR with "[WIP]" at the start of the description. The CI system will not merge it until the "[WIP]" is removed and all tests are passing.
+
+## Setting up a test cluster
+
+Should work with any k8s cluster; tested with AWS and [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/download). See below for extra steps with some specific cluster providers.
+
+Cluster creation requires a pull-secret from the cluster provider. You need
+to add a secret to this file so `make` can pull images from the openshift
+CI registry:
+
+Copy the `oc login` command from https://api.ci.openshift.org/oauth/token/request, then:
+```
+oc login api.ci.openshift.org ... # command from the web page
+oc registry login --to my-secret-file
+oc logout
+# This creates or adds to my-secret-file
+```
+
+Once created, you can `export KUBECONFIG=/path/to/my/cluster/config` or copy your config to `$HOME/.kube/config`.
+
+**Note**: Don't symlink `$HOME/.kube/config`. Some commands may overwrite this file and break the symlink, causing much confusion.
+
+Log in to your cluster as a user with the `cluster-admin` role. User `kubeadmin` is usually predefined with this role.
```

Comment:
My experience has been this is not correct.  You have to explicitly grant that role which seems strange.  You can confirm this when logging into the kibana interface and it does not think you are an admin

Reply:
All this stuff taken out of this PR, will do a separate PR to improve the HACKING doc.